patches/components-browser_sync-profile_sync_service.h.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/components/browser_sync/profile_sync_service.h b/components/browser_sync/profile_sync_service.h
+index d97259718bcabbe417c9e5c91bc6cfd454755020..92e7179fd191352057a6b8b05f2a71b7ee569c49 100644
+--- a/components/browser_sync/profile_sync_service.h
++++ b/components/browser_sync/profile_sync_service.h
+@@ -44,6 +44,7 @@
+ #include "components/version_info/version_info.h"
+ #include "google_apis/gaia/google_service_auth_error.h"
+ #include "url/gurl.h"
++#include "brave/components/sync/driver/brave_sync_service_base.h"
+ 
+ class SigninManagerWrapper;
+ 
+@@ -157,7 +158,8 @@ class SyncAuthManager;
+ //   Once first setup has completed and there are no outstanding
+ //   setup-in-progress handles, CanConfigureDataTypes() will return true and
+ //   datatype configuration can begin.
+-class ProfileSyncService : public syncer::SyncService,
```

Comment:
why are we making ProfileSyncService subclass BraveSyncServiceBase instead of the other way around?

Reply:
if BraveSyncService subclasses ProfileSyncService then we can use the preprocessor to override it

Reply:
the only purpose of this subclassing is to catch the point when I can create brave_sync::ControllerImpl . I will find a better way.